### DISCUSSION TITLE
go.mod wrong?

### DISCUSSION THREAD

**Discussion by nicpottier** [carlmjohnson/flowmatic#11] [Support] [answered]
Good looking library, but go doesn't like that the go.mod doesn't match the github repo. Just FYI.

```➜  % go get github.com/earthboundkid/flowmatic
go: downloading github.com/earthboundkid/flowmatic v0.23.4
go: github.com/earthboundkid/flowmatic@upgrade (v0.23.4) requires github.com/earthboundkid/flowmatic@v0.23.4: parsing go.mod:
	module declares its path as: github.com/carlmjohnson/flowmatic
	        but was required as: github.com/earthboundkid/flowmatic
```

**Comment by nicpottier** [comment]
Adding this to my `go.mod` fixed it FWIW:

```
replace github.com/earthboundkid/flowmatic v0.23.4 => github.com/carlmjohnson/flowmatic v0.23.4
```

**Comment by earthboundkid** [comment, marked as answer]
The URL of the repo in Github is "wrong". The import path is github.com/carlmjohnson/flowmatic. See https://gist.github.com/earthboundkid/8915002ae0e531cecdfc58bc6453ac80.

**Comment by nicpottier** [comment]
This may be worth sticking in the README, it definitely hung me up for a second.

Also FWIW this does feel like a big enough hoop for adoption that making a breaking major version change is worth it. (or if you don't want to enter `>1` Go land of API contrats then break the rules and keep in `0.*`. But future users > past users and this is almost certainly causing people unnecessary friction. (vs a one time friction of current projects having to rename their imports when they update)

**Reply by earthboundkid** [reply]
The import URL will probably be changed if I finish https://github.com/earthboundkid/flowmatic/pull/9 which changes the API to use generic iterators. That needs more thought first though.
